packages/kbn-interpreter/src/public/interpreter.js

```diff
@@ -52,10 +52,19 @@ export async function initializeInterpreter(socket, typesRegistry, functionsRegi
   });
 
   // Create the function list
+  let gotFunctionList = false;
   socket.once('functionList', (fl) => {
+    gotFunctionList = true;
     resolve(fl);
   });
-  socket.emit('getFunctionList');
+
+  const interval = setInterval(() => {
```

Comment:
not sure why, but sometimes (everytime in canvas some test actually) the server will not respont to initial getFunctionList request ..

Reply:
You mean the `functionList` event is never fired, or sometimes the value if gets is an empty array?

Reply:
I put some console.log's in there, and I only see the event fire once, and the handler doesn't have a problem with it.

![screenshot 2019-01-04 16 19 20](https://user-images.githubusercontent.com/404731/50715936-96776500-103c-11e9-9afb-b0263fe8ea56.png)

Reply:
yeah to reproduce, remove this change (so just socket.emit('getFunctionList') 
and then run selenium tests for the canvas smoke test ....

server never receives the getFunctionList event ... 
so it might be that the socket on server side is not yet ready ? weird ...

Reply:
Interesting. Maybe there's a race condition that only comes up on resource-constrained systems...